controllers/virtinkmachine_controller.go

```diff
@@ -148,6 +153,28 @@ func (r *VirtinkMachineReconciler) reconcile(ctx context.Context, machine *infra
 			return fmt.Errorf("bootstrap data is nil")
 		}
 
+		dataVolumes := r.buildDataVolumes(ctx, machine)
+		for _, dataVolume := range dataVolumes {
+			dataVolumeKey := types.NamespacedName{
+				Namespace: dataVolume.Namespace,
+				Name:      dataVolume.Name,
+			}
+			dataVolumeNotFound := false
+			createdDataVolume := cdiv1beta1.DataVolume{}
+			if err := r.Get(ctx, dataVolumeKey, &createdDataVolume); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return fmt.Errorf("get DataVolume: %s", err)
+				}
+				dataVolumeNotFound = true
+			}
+			if dataVolumeNotFound {
+				if err := infraClusterClient.Create(ctx, dataVolume); err != nil {
```

Comment:
Clean the DataVolume when deleting VirtinkMachine?

Reply:
I'm not sure delete DataVolume automatically  is a good choice or not, as DataVolume is for persistent data. So I followed the k8s [Statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations), when deleting it or scaling down will not delete volumes to keep data safety.